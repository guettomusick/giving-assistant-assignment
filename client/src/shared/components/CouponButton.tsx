import React, { FC } from 'react';
import styled from 'styled-components';

import Button from './Button';
import Thumbnail from './Thumbnail';
import { CmsImage } from '../types/cms';

const CouponButtonContainer = styled.div`
  display: flex;
  flex-direction: column;
  cursor: pointer;
  &:hover {
    opacity: .7;
  }
`;

const StyledThumbnail = styled(Thumbnail)`
  border-radius: 6px 6px 0 0;
`;

type StyledButtonProps = {
  hasThumbnail: boolean;
};

const StyledButton = styled(Button)<StyledButtonProps>`
  ${props => props.hasThumbnail && `
    border-radius: 0 0 6px 6px;
  `};
`;

type Props = {
  title: string,
  onClick: () => any,
  thumbnail?: CmsImage,
};

const CouponButton: FC<Props> = ({
  title,
  onClick,
  thumbnail,
}) => (
  <CouponButtonContainer>
    { thumbnail && (
      <StyledThumbnail
        src={ `/api${thumbnail.formats?.medium?.url || thumbnail.url}` } 
        onClick={ onClick }
      />
    )}
    <StyledButton
      onClick={ onClick }
      hasThumbnail = { !!thumbnail }
    >
      { title }
    </StyledButton>
  </CouponButtonContainer>
);

export default CouponButton;