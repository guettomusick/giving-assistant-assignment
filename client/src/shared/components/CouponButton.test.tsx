import React from 'react';
import renderer from 'react-test-renderer';

import CouponButton from './CouponButton';

describe('Button', () => {
  it('renders', () => {
    const component = renderer.create(
      <CouponButton title='Button' onClick={ () => {} }/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
