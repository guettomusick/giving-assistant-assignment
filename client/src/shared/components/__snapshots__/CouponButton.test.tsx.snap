// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CouponButton renders 1`] = `
<div
  className="sc-AxiKw glNTBB"
>
  <button
    className="sc-AxjAm sc-AxhUy fHieOr"
    onClick={[Function]}
  >
    Button
  </button>
</div>
`;
